integration-tests/tests/common/mod.rs

```diff
@@ -76,17 +84,29 @@ impl IntegrationTest {
             .arg(self.code)
             .arg("--shim")
             .arg(self.shim)
+            .stdin(Stdio::piped())
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
             .spawn()
             .expect("failed to run the test");
 
+        let input = cmd.stdin.as_mut().unwrap();
+        input
+            .write_all(stdin.as_ref())
+            .expect("Failed to write stdin");
         let ecode = match cmd.wait_timeout(seconds).unwrap() {
             Some(status) => status.code(),
             None => {
                 cmd.kill().unwrap();
                 panic!("killed by watchdog!");
             }
         };
+        let output = cmd
+            .wait_with_output()
+            .expect("Failed to read stdout/stderr");
 
+        assert_eq!(stdout.as_ref().to_vec(), output.stdout);
+        assert_eq!(stderr.as_ref().to_vec(), output.stderr);
```

Comment:
```suggestion
        assert_eq!(stdout.as_ref(), output.stdout.as_ref());
        assert_eq!(stderr.as_ref(), output.stderr.as_ref());
```

We don't want to use `to_vec()` here. `to_vec()` will allocate memory. Instead, we express the existing `Vec` as a slice using `as_ref()`, which doesn't allocate memory.

Reply:
Yeah it makes sense to not allocate memory.
I am using as_slice to convert it from vec --> slice
 as_ref() is resulting in an error